lib/screens/add_balance_screen.dart

```diff
@@ -0,0 +1,160 @@
+import 'package:e_wallet/models/user_model.dart';
+import 'package:e_wallet/providers/transaction_provider.dart';
+import 'package:e_wallet/providers/user_provider.dart';
+import 'package:e_wallet/util/custom_format_util.dart';
+import 'package:e_wallet/util/decimal_input_formatter.dart';
+import 'package:e_wallet/validators/add_balance_validator.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:intl/intl.dart';
+import 'package:flutter/services.dart';
+
+class AddBalance extends StatefulWidget {
+  final String routeName = '/add-balance';
+
+  @override
+  _AddBalanceState createState() => _AddBalanceState();
+}
+
+class _AddBalanceState extends State<AddBalance> {
+  final _addBalanceFormKey = GlobalKey<FormState>();
+  final oCcy = new NumberFormat("#,###.00", "en_US");
+  TextEditingController _amountController = TextEditingController();
+  TextEditingController _descriptionController = TextEditingController();
+  bool isLoading = false;
+  int amountIntValue = 0;
+
+  @override
+  Widget build(BuildContext context) {
+    UserModel user = Provider.of<UserProvider>(context).userModel;
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Add Balance'),
+        elevation: 0,
+      ),
+      body: Builder(builder: (context) {
+        return Padding(
+          padding: EdgeInsets.all(20),
+          child: IntrinsicWidth(
+            stepWidth: double.infinity,
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              children: <Widget>[
+                Row(
+                  children: [
+                    Text('Current Balance:'),
+                    if (user != null) ...[
```

Comment:
You may rewrite this as:
```
Text((user != null)?
        CustomFormatUtil().formatIntAsCurrencyAdd(
          user.balance,
          amountIntValue,
        ) : '',
      )
```

Reply:
Nice version because I could add "0" as value when user is null.